apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -1029,6 +1032,57 @@ receive_maximum(ConnInfo) ->
 expiry_interval(ConnInfo) ->
     maps:get(expiry_interval, ConnInfo, 0).
 
+%%--------------------------------------------------------------------------------
+%% `emqx_config_handler' API
+%%--------------------------------------------------------------------------------
+
+-define(ROOT_KEY, session_persistence).
+-define(CACHE_PREFETCH_PATH, [?ROOT_KEY, cache_prefetch_topic_filters]).
+-define(CACHE_GC_INTERVAL_PATH, [?ROOT_KEY, cache_gc_interval]).
+
+post_config_update(
+    ?CACHE_PREFETCH_PATH, _Req, NewConf, OldConf, _AppEnv
+) ->
+    handle_update_cache_prefetch_topic_filters(NewConf, OldConf),
+    ok;
+post_config_update(
+    ?CACHE_GC_INTERVAL_PATH, _Req, NewConf, _OldConf, _AppEnv
+) ->
+    handle_update_cache_gc_interval(NewConf),
+    ok;
+post_config_update(_Path, _Req, _NewConf, _OldConf, _AppEnv) ->
+    ok.
+
+propagated_post_config_update(
+    ?CACHE_PREFETCH_PATH, _Req, NewConf, OldConf, _AppEnvs
+) ->
+    handle_update_cache_prefetch_topic_filters(NewConf, OldConf),
+    ok;
+propagated_post_config_update(
+    ?CACHE_GC_INTERVAL_PATH, _Req, NewConf, _OldConf, _AppEnvs
+) ->
+    handle_update_cache_gc_interval(NewConf),
+    ok;
+propagated_post_config_update(_Path, _UpdateReq, _NewConf, _OldConf, _AppEnvs) ->
+    ok.
+
+handle_update_cache_prefetch_topic_filters(NewTopics, OldTopics) ->
+    Added = lists:usort(NewTopics -- OldTopics),
+    Removed = lists:usort(OldTopics -- NewTopics),
+    emqx_ds_cache:remove_cached_topic_filters(?PERSISTENT_MESSAGE_DB, Removed),
```

Comment:
Seems that the config is for persistent sessions in general;
However it configures `emqx_ds_cache` which belongs to a concrete backend (replication layer)

Reply:
Maybe we could somehow isolate `emqx_ds` universal code from concrete "main" implementation 😓 Because otherwise it is unclear that some shortcuts from universal code to the replication backend break `emqx_ds` interface 🤔

Reply:
I think that the PS DS cache may be the same for all backends, since the DB is the same.  There's no need to differentiate between the backends in this case.  

For example, FDB may just do the same wrapping as replication_layer does.  I did it in replication layer because the backend needs to implement `last_seen_key` in order for the cache to work properly.

Reply:
>in replication layer because the backend needs to implement last_seen_key in order for the cache to work properly.

🤔  But we have `last_seen_key` in the `emqx_ds` behaviour... so we can be sure that it is implemented from the `emqx_ds` layer

Reply:
it's an optional callback: https://github.com/emqx/emqx/pull/12171/files#diff-c7f537324b94a1e8f072e809aeb185a9adda2c89d8aaea78438c3aa4e5c5f4c1R151

Reply:
some DBs/backends might not be able to implement it; for example, a backend for retained messages might not implement it, as iterations will probably be done by scanning all topics matching a filter without any strict notion of time/order.